src/loader.c

```diff
@@ -157,8 +159,25 @@ static void *imv_loader_bg_new_img(void *data)
   pthread_mutex_lock(&ldr->lock);
   char *path = strdup(ldr->path);
   pthread_mutex_unlock(&ldr->lock);
-
-  FREE_IMAGE_FORMAT fmt = FreeImage_GetFileType(path, 0);
+  FREE_IMAGE_FORMAT fmt = NULL;
+  FIBITMAP *bmp = NULL;
+  if(ldr->stdin_buffer != NULL && strcmp(path, "stdin") == 0){
```

Comment:
At this point, we should only do the `"stdin"` and `stdin_buffer` check and use that to set a variable, something like `int load_from_memory`.

Then each time we're touching `path` via FreeImage check `load_from_memory` and use that to decide between using `path`, and using `stdin_buffer.`

Doing it that way will simplify the logical changes made whilst letting animated gifs work with the stdin buffer.